Type unexpectedly gains const in !?T -> ?T -> *T capture chain

When writing https://github.com/ziglang/zig/pull/5818 I wanted to add the following in order to match the other test cases:

```zig
   // Access the value by reference using a pointer capture.
   var d: anyerror!?u32 = 3;
   if (d) |optional_value| {
       if (optional_value) |*value| {
           value.* = 9;
       }
   } else |err| {
       unreachable;
   }

   if (d) |optional_value| {
       assert(optional_value.? == 9);
   } else |err| {
       unreachable;
   }
```

but it fails to compile with:

```
.\if-else-catch.zig:80:23: error: cannot assign to constant
            value.* = 9;
                      ^
```

After adding some `@compileLog`'s I found that `optional_value` is of type `?u32` (expected) but the `*value` capture is of type `*const u32` (unexpected)

The following works correctly and doesn't inject a const:

```zig
   var e: ?u32 = 3;
   if (e) |*value| {
       value.* = 2;
   }
```

--- a/doc/langref.html.in
+++ b/doc/langref.html.in
@@ -3886,6 +3886,22 @@ test "if error union with optional" {
     } else |err| {
         assert(err == error.BadValue);
     }
+
+    // Access the value by reference by using a pointer capture each time.
+    var d: anyerror!?u32 = 3;
+    if (d) |*optional_value| {
+        if (optional_value.*) |*value| {
+            value.* = 9;
+        }
+    } else |err| {
+        unreachable;
+    }
+
+    if (d) |optional_value| {
+        assert(optional_value.? == 9);
+    } else |err| {
+        unreachable;
+    }
 }
       {#code_end#}
       {#see_also|Optionals|Errors#}

